src/pkceUtils.ts

```diff
@@ -24,6 +24,11 @@ export function generateRandomString(length: number): string {
 export async function generateCodeChallenge(codeVerifier: string): Promise<string> {
   const encoder = new TextEncoder()
   const bytes: Uint8Array = encoder.encode(codeVerifier) // Encode the verifier to a byteArray
+  if (!window.crypto.subtle?.digest) {
+    throw new Error(
+      "The context/environment is not secure, and does not support the 'crypto.subtle' module. See: https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle for details"
+    )
+  }
```

Comment:
I guess you could check this in the start of the function, no? No need to make an encoder or encode the verifier if we're just gonna throw an exception.